src/networking/LanCommunicator.java

```diff
@@ -1,27 +1,186 @@
 package networking;
 /**
+ * This file contains the class LanCommunicator which is the object that other modules
+ * use to start the communication , send the messages across the network and subscribe
+ * for messages that they should be receiving.
  *
  * @author Madaka Srikar Reddy
  */
 
-import queueManagement.*;
 
-class LanCommunicator implements ICommunicator{
+import java.util.HashMap;
+import networking.queueManagement.*;
+import networking.utility.*;
+import infrastructure.validation.logger.*;
 
+public class LanCommunicator implements ICommunicator{
+
+    /** SendQueueListener that will be listening on the send Queue */
+    private SendQueueListener sendQueueListener;
+    private Thread sendQueueListenerWorker;
+
+    /** NetworkListener that will be listening on the network */
+    private SocketListener socketListener;
+    private Thread socketListenerWorker;
+
+    /** ProcessingReceiveQueueListener that will be listening on the Processing Receive Queue */
+    private ReceiveQueueListener processingReceiveQueueListener;
+    private Thread processingReceiveQueueListenerWorker;
+
+    /** ContentReceiveQueueListener that will be listening on the Content Receive Queue */
+    private ReceiveQueueListener contentReceiveQueueListener;
+    private Thread contentReceiveQueueListenerWorker;
+
+    /** Send Queue that has the message that should be sent across the network */
+    private IQueue<OutgoingPacket> sendQueue;
+
+    /** Receive Queue in which we have the messages that needs to be sent to the processing module */
+    private IQueue<IncomingPacket> processingReceiveQueue;
+
+    /** Receive queue in which we have the messages that needs to be sent to the content module */
+    private IQueue<IncomingPacket> contentReceiveQueue;
+
+    /** Hashmap contains the handlers for the respective modules processing and content */
+    private HashMap<String, INotificationHandler> handlerMap;
+    
+    /** Port Number that we will be listening on the network */
+    private int portNumber;
+    
+    /** Boolean check variable for listeners to use */
+    private static boolean isRunning = false;
+    
+    /** Logger for logging errors and activities */
+    ILogger logger = LoggerFactory.getLoggerInstance();
+
+    /**
+     * constructor that takes port and also initializes a hashmap
+     * 
+     * @param port that we will be passing to the socketListener
+     */
+    public LanCommunicator(int port) {
+    	portNumber = port;
+        handlerMap = new java.util.HashMap<>();
```

Comment:
change java.util.HashMap, it is redundant

Reply:
fixed the requested changes